lib/ansible/modules/remote_management/cpm/cpm_serial_port_info.py

```diff
@@ -0,0 +1,179 @@
+description:
+    - "Get Serial port parameters from WTI OOB and PDU devices"
+options:
+  cpm_url:
+    description:
+      - This is the URL of the WTI device to send the module.
+    required: true
+  cpm_username:
+    description:
+      - This is the Username of the WTI device to send the module.
+    required: true
+  cpm_password:
+    description:
+      - This is the Password of the WTI device to send the module.
+    required: true
+  use_https:
+    description:
+      - Designates to use an https connection or http connection.
+    required: false
+    type: bool
+    default: true
+  validate_certs:
+    description:
+      - If false, SSL certificates will not be validated. This should only be used
+      - on personally controlled sites using self-signed certificates.
+    required: false
+    type: bool
+    default: true
+  use_proxy:
+    description: Flag to control if the lookup will observe HTTP proxy environment variables when present.
+    required: false
+    type: bool
+    default: false
+  port:
+    description:
+      - This is the serial port number that is getting retrieved. It can include a single port
+      - number, multiple port numbers separated by commas or an '*' character for all ports.
+    required: true
+"""
+
+EXAMPLES = """
+# Get Parameters for one Serial Port
+- name: Get the Serial Port Parameters for port 2 of a WTI device
+  cpm_serial_port_info:
+    cpm_url: "nonexist.wti.com"
+    cpm_username: "super"
+    cpm_password: "super"
+    use_https: true
+    validate_certs: false
+    port: "2"
+
+# Get Parameters for multiple Serial Ports
+- name: Get the Serial Port Parameters for ports 2 and 4 of a WTI device
+  cpm_serial_port_info:
+    cpm_url: "nonexist.wti.com"
+    cpm_username: "super"
+    cpm_password: "super"
+    use_https: true
+    validate_certs: false
+    port: "2,4"
+
+# Get Parameters for All Serial Ports
+- name: Get the Serial Port Parameters for all ports of a WTI device
+  cpm_serial_port_info:
+    cpm_url: "nonexist.wti.com"
+    cpm_username: "super"
+    cpm_password: "super"
+    use_https: true
+    validate_certs: false
+    port: "*"
+"""
+
+RETURN = """
+data:
+    description: The output JSON returned from the commands sent
+    returned: always
+    type: str
+"""
+
+import base64
+import json
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_text, to_bytes, to_native
+from ansible.module_utils.six.moves.urllib.error import HTTPError, URLError
+from ansible.module_utils.urls import open_url, ConnectionError, SSLValidationError
+
+
+def run_module():
+    # define the available arguments/parameters that a user can pass to
+    # the module
+    module_args = dict(
+        cpm_url=dict(type='str', required=True),
+        cpm_username=dict(type='str', required=True),
+        cpm_password=dict(type='str', required=True, no_log=True),
+        port=dict(type='str', required=True),
```

Comment:
Since this accepts a list of ports, change the type to `list`. This requires it to be transformed a bit, but would be a better UI.

A `list` type parameter will accept a comma separated string as well as single items, which field validation will return as a string.